apps/api/src/services/user.service.ts

```diff
@@ -83,28 +84,30 @@ export class UserService {
 
 	async updateUser(authUser: User, updateUserDto: UpdateUserDto) {
 		try {
-			const user = await this.prismaService.user.update({
-				where: {
-					uid: authUser.id,
-				},
-				data: {
-					slug: updateUserDto.slug ? updateUserDto.slug : authUser.slug,
-					photoURL: updateUserDto.photoUrl
-						? updateUserDto.photoUrl
-						: authUser.photo_url,
-					displayName: updateUserDto.displayName
-						? updateUserDto.displayName
-						: authUser.display_name,
-					isStudent: Object.hasOwn(updateUserDto, 'isCollegeStudent')
-						? updateUserDto.isCollegeStudent
-						: authUser.is_student,
-					collegeName: updateUserDto.collegeName || authUser.college_name,
+			const data = {
+				slug: updateUserDto.slug ? updateUserDto.slug : authUser.slug,
+				photo_url: updateUserDto.photoUrl
+					? updateUserDto.photoUrl
+					: authUser.photo_url,
+				display_name: updateUserDto.displayName
+					? updateUserDto.displayName
+					: authUser.display_name,
+				is_student: Object.hasOwn(updateUserDto, 'isCollegeStudent')
+					? updateUserDto.isCollegeStudent
+					: authUser.is_student,
+				college_name: updateUserDto.collegeName || authUser.college_name,
+			};
+
```

Comment:
No need to add the other data here. The zod validator will throw error if unwanted data is specified